numpy/core/tests/test_casting_unittests.py

```diff
@@ -147,6 +147,9 @@ def test_to_void(self):
         assert not np.can_cast("U1", "V1")
         # Structured to unstructured is just like any other:
         assert np.can_cast("d,i", "V", casting="same_kind")
+        # Unstructured void to unstructured is actually no cast at all:
+        assert np.can_cast("V3", "V", casting="no")
+        assert np.can_cast("V0", "V", casting="no")
```

Comment:
If this is confusing, it is because `V` is interpreted as the *unsized* void (or void DType) and not `V0` here.  You can't use an actual `V0` here, but if you could the result would be same-kind.